rootfs/api/models/app.py

```diff
@@ -331,15 +331,21 @@ def scale(self, user, structure):  # noqa
                     'Container type {} does not exist in application'.format(container_type))
 
         # merge current structure and the new items together
-        new_structure = self.structure.copy()
+        old_structure = self.structure
+        new_structure = old_structure.copy()
         new_structure.update(structure)
 
         if new_structure != self.structure:
             # save new structure to the database
             self.structure = new_structure
             self.save()
 
-            self._scale_pods(structure)
+            try:
+                self._scale_pods(structure)
+            except ServiceUnavailable:
+                # scaling failed, go back to old scaling numbers
```

Comment:
`_scale_pods` takes `KubeException` (and by exension `KubeHTTPException`) and wraps it into a `ServiceUnavailable`

Prior to this change the call to `_scale_pod` would let `ServiceUnavailable` bubble all the way up